packages/mask/src/plugins/Avatar/hooks/useNFT.ts

```diff
@@ -35,7 +36,13 @@ export function useNFT(
         const amount = asset?.priceInToken?.amount ?? asset?.price?.[CurrencyType.USD] ?? '0'
         const name = contract?.name || metadata?.name || ''
         const imageURL = metadata?.imageURL
-        const permalink = asset?.link ?? Others?.explorerResolver.nonFungibleTokenLink(chainId, address, tokenId)
+        let permalink = asset?.link
+        if (!permalink) {
+            permalink =
+                pluginId === NetworkPluginID.PLUGIN_SOLANA
+                    ? MagicEdenAPI.getUrl(tokenId)
```

Comment:
Recommend composing the `asset?.link` in `web3-provider`, and you don't need to export `MagicEdenAPI` anymore.

Reply:
> Recommend composing the asset?.link in web3-provider

Already did there.

In this case, we need a webpage to view the NFT, for EVM, Etherscan provides such webpage, while Solana explroer does not, so we show Magiceden's page instead.

Reply:
The `getUrl` is a private API. Using a private API makes the UI less portable. We should add a thing like `getNonFungibleTokenLink` instead of using a private API.